File: c/validators/writer/test_quarter_circle.c
// SPDX-License-Identifier: MIT
#include "test_utils.h"

static bool validate_quarter_circle(const char* directory) {
    return validate_single_arc("quarter_circle", directory, "quarter_circle.skp",
                               6, 10.0, 1.5707963267948966, false);
}

WRITER_VALIDATOR_MAIN("quarter_circle", validate_quarter_circle)

File: python/writer/generators/test_quarter_circle.py
# SPDX-License-Identifier: MIT
import math
from pathlib import Path

from .common import run


def generate(skppy, destination: Path) -> None:
    model = skppy.new_model()
    model.entities.add_arc_curve((0, 0, 0), (0, 0, 1), 10.0, 0.0, math.pi / 2.0, 6)
    skppy.save(model, destination)


if __name__ == "__main__":
    raise SystemExit(run(generate, "quarter_circle"))
